internal/objects/validator_test.go

```diff
@@ -752,8 +780,13 @@ func TestCreatingConflictSource(t *testing.T) {
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
 			// Setup
+			r := &Reconciler{
```

Comment:
Maybe say why you're adding this - e.g. because the validator needs to know the propagation mode.

Do you know why this ever worked before?

Reply:
I believe it worked previously because before there was only one mode that allowed propagation, so if it it was [checking for conflicts](https://github.com/kubernetes-sigs/hierarchical-namespaces/blob/347c5a3f92463d1157650172524d7743bdc6faa7/internal/objects/validator.go#L329-L357), it could assume it's `Propagate` mode since any other mode would have [already returned](https://github.com/kubernetes-sigs/hierarchical-namespaces/blob/347c5a3f92463d1157650172524d7743bdc6faa7/internal/objects/validator.go#L85-L86). Now that we've added another mode that allows propagation under certain conditions then we need to differentiate and therefore set up a reconcile with a specific mode